src/Symfony/Component/Form/FormOrderer.php

```diff
@@ -0,0 +1,327 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form;
+
+use Symfony\Component\Form\Exception\InvalidConfigurationException;
+
+/**
+ * Form orderer.
+ *
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+class FormOrderer
+{
+    /**
+     * @var array
+     */
+    private $weights;
+
+    /**
+     * @var array
+     */
+    private $deferred;
+
+    /**
+     * @var int
+     */
+    private $firstWeight;
+
+    /**
+     * @var int
+     */
+    private $currentWeight;
+
+    /**
+     * @var int
+     */
+    private $lastWeight;
+
+    /**
+     * Orders the form.
+     *
+     * @param FormInterface $form The form.
+     *
+     * @return array The ordered form child names.
+     *
+     * @throws \Symfony\Component\Form\Exception\InvalidConfigurationException If a position is not valid.
```

Comment:
You should add a use statement instead of using the FQCN here. And remove the dot at the end